metadata-ingestion/src/datahub/ingestion/source/gc/dataprocess_cleanup.py

```diff
@@ -265,13 +267,17 @@ def keep_last_n_dpi(
                     self.report.report_failure(
                         f"Exception while deleting DPI: {e}", exc=e
                     )
-            if deleted_count_last_n % self.config.batch_size == 0:
+            if (
+                deleted_count_last_n % self.config.batch_size == 0
+                and deleted_count_last_n > 0
+            ):
                 logger.info(f"Deleted {deleted_count_last_n} DPIs from {job.urn}")
                 if self.config.delay:
                     logger.info(f"Sleeping for {self.config.delay} seconds")
                     time.sleep(self.config.delay)
 
-        logger.info(f"Deleted {deleted_count_last_n} DPIs from {job.urn}")
+        if deleted_count_last_n > 0:
+            logger.info(f"Deleted {deleted_count_last_n} DPIs from {job.urn}")
```

Comment:
It seems we are repeating code here, how about refactoring **both** ifs to
```suggestion
        if deleted_count_last_n > 0:
            logger.info(f"Deleted {deleted_count_last_n} DPIs from {job.urn}")
            if deleted_count_last_n % self.config.batch_size == 0 and self.config.delay:
                    logger.info(f"Sleeping for {self.config.delay} seconds")
                    time.sleep(self.config.delay)              
```